t/058-tcp-socket.t

```diff
@@ -3808,3 +3808,131 @@ received:
 received: truefalsenil
 --- no_error_log
 [error]
+
+
+
+=== TEST 64: *b pattern for receive
```

Comment:
I suggest you also duplicate this test case in a separate `.t` test file which *requires*` the use of mockeagain reading test mode for the extreme case of receiving one byte at a time:

```
t/058-tcp-socket.t .. 115/416
#   Failed test 'TEST 64: *b pattern for receive - response_body - response is expected (repeated req 0, req 0)'
#   at /home/agentzh/git/lua-nginx-module/../test-nginx/lib/Test/Nginx/Socket.pm line 1555.
# @@ -1,3 +1,14 @@
#  1
# -22
# +2
# -hello world
# +2
# +h
# +e
# +l
# +l
# +o
# +
# +w
# +o
# +r
# +l
# +d

#   Failed test 'TEST 64: *b pattern for receive - response_body - response is expected (repeated req 1, req 0)'
#   at /home/agentzh/git/lua-nginx-module/../test-nginx/lib/Test/Nginx/Socket.pm line 1555.
# @@ -1,3 +1,14 @@
#  1
# -22
# +2
# -hello world
# +2
# +h
# +e
# +l
# +l
# +o
# +
# +w
# +o
# +r
# +l
# +d
t/058-tcp-socket.t .. 379/416 # Looks like you failed 2 tests of 416.
```

Reply:
Oh, and maybe mockeagain *writing* mode too :)

Reply:
@agentzh 
Additional test added.